packages/agents/src/agent-core/llm/chat/Chat.ts

```diff
@@ -27,56 +32,71 @@ export class Chat {
     return this._chatLogs.messages;
   }
 
-  public add(type: ChatLogType, msg: ChatMessage | ChatMessage[]) {
+  public async add(type: ChatLogType, msg: ChatMessage | ChatMessage[]): Promise<void> {
     let msgs = Array.isArray(msg) ? msg : [msg];
 
-    for (const msg of msgs) {
+    const msgsWithTokens = msgs.map((msg) => {
```

Comment:
Could you just get tokens? 
Since you're never using this joint type and for `msgsToAdd` you can just use `msgs`